addons/sale/wizard/sale_order_cancel_views.xml

```diff
@@ -5,14 +5,46 @@
         <field name="model">sale.order.cancel</field>
         <field name="arch" type="xml">
             <form>
-                <field name="order_id" invisible="1"/>
-                <field name="display_invoice_alert" invisible="1"/>
-                <div attrs="{'invisible': [('display_invoice_alert', '=', False)]}">
-                    Draft invoices for this order will be cancelled.
-                </div>
+                <group col="1">
+                    <field name="render_model" invisible="1"/>
+                    <field name="order_id" invisible="1"/>
+                    <field name="template_id" invisible="1"/>
+                    <field name="display_invoice_alert" invisible="1"/>
+                    <div col="2"
+                         class="alert alert-warning"
+                         role="alert">
+                        <span>Are you sure you want to cancel this order? <br/></span>
+                        <span id="display_invoice_alert"
+                              attrs="{'invisible': [('display_invoice_alert', '=', False)]}">
+                            Draft invoices for this order will be cancelled. <br/>
+                        </span>
+                    </div>
+                    <group col="2">
+                        <field name="recipient_ids"
+                               widget="many2many_tags_email"
+                               context="{'force_email':True,
+                                         'show_email':True,
+                                         'no_create_edit': True}"/>
+                    </group>
+                    <group col="2">
+                        <field name="subject" placeholder="Subject"/>
+                    </group>
+                    <field name="body"
+                           class="oe-bordered-editor"
+                           options="{'style-inline': true}"/>
+                </group>
                 <footer>
-                    <button string="Confirm" name="action_cancel" type="object" class="oe_highlight" data-hotkey="q"/>
-                    <button string="Cancel" class="btn btn-default" special="cancel" data-hotkey="z"/>
```

Comment:
Why did you remove the hotkeys specification ?

Reply:
2 buttons => 3, it was unclear which shortcut should be on which button, we let the auto-assigned buttons instead.